beacon-chain/p2p/peers/status.go

```diff
@@ -0,0 +1,311 @@
+// Package peers provides information about peers at the Ethereum protocol level.
+// "Protocol level" is the level above the network level, so this layer never sees or interacts with (for example) hosts that are
+// uncontactable due to being down, firewalled, etc.  Instead, this works with peers that are contactable but may or may not be of
+// the correct fork version, not currently required due to the number of current connections, etc.
+//
+// A peer can have one of a number of states:
+//
+// - disconnected if we are not able to talk to the remote peer
+// - connecting if we are attempting to be able to talk to the remote peer
+// - connected if we are able to talk to the remote peer
+// - disconnecting if we are attempting to stop being able to talk to the remote peer
```

Comment:
Nit: sort these statuses alphabetically

Reply:
Edit: i see why you sorted them this way (same order as ID), but maybe is better for readability in godocs? Up to you/

Reply:
Yeah sorting alphabetically make sense.  Will fix.

Reply:
Alphabetically sorted in latest commit.